File: JavaScript/0049_group_anagrams/solution.js
function group_anagrams(strs) {
    let result = {};
    for (let string of strs) {
        let canonical = string.split("").sort().join("");
        if (canonical in result) {
            result[canonical].push(string);
        } else {
            result[canonical] = [string];
        }
    }
    return Object.values(result);
}

const strs = ["eat","tea","tan","ate","nat","bat"];
console.log(group_anagrams(strs));

File: Python/0049_group_anagrams/solution.py
def group_anagrams(strs):
    result = {}
    for string in strs:
        canonical = "".join(sorted(string))
        if canonical in result:
            result[canonical].append(string)
        else:
            result[canonical] = [string]
    return list(result.values())

strs = ["eat","tea","tan","ate","nat","bat"]
print(group_anagrams(strs))